src/Broadcast.chpl

```diff
@@ -135,4 +145,39 @@ module Broadcast {
     expandedVals = (+ scan expandedVals);
     return (expandedVals == 1);
   }
+
+  proc broadcast(segs: [?sD] int, segString: borrowed SegString, size: int) throws {
+    ref segOff = segString.offsets.a;
+    const high = sD.high;
+    var strSize: int;
+    var diffs: [sD] int;
+
+    forall (i, d, seg, off) in zip (sD, diffs, segs, segOff) with (+ reduce strSize) {
+      if i == high {
+        strSize += (size - seg) * (segString.nBytes - off);
+        d = segString.nBytes - off;
+      } else {
+        strSize += (segs[i+1] - seg) * (segOff[i+1] - off);
+        d = segOff[i+1] - off;
+      }
+    }
+
+    var broadDist = broadcast(segs, diffs, size);
+    var offDiff = makeDistArray(size, int);
+    offDiff[1..] = broadDist[..size-2];
+
+    var offsets = (+ scan offDiff);
```

Comment:
does this just end up being an exclusive scan? i.e. equivalent to
```chapel
var broadDist = broadcast(segs, diffs, size);
var offsets = (+ scan broadDist) - broadDist;
```

Reply:
Yes, good catch!